tests/test_multiaddr.py

```diff
@@ -218,3 +223,47 @@ def test_get_value():
     assert_value_for_proto(a, P_IP4, "0.0.0.0")
     assert_value_for_proto(a, P_UDP, "12345")
     assert_value_for_proto(a, P_UTP, "")
+
+
+def test_bad_initialization_no_params():
+    with pytest.raises(ValueError):
+        Multiaddr()
+
+
+def test_bad_initialization_too_many_params():
+    with pytest.raises(ValueError):
+        Multiaddr("/ip4/0.0.0.0", string_to_bytes("/ip4/0.0.0.0"))
+
+
+def test_get_value_too_many_fields_protocol(monkeypatch):
+    monkeypatch.setattr("multiaddr.multiaddr.Multiaddr.__str__",
+                        lambda ignore: '/udp/1234/5678')
```

Comment:
I'm not quite sure I understand this test. It looks like you're patching `Multiaddr.__str__` to return an invalid representation (and ignoring the actual constructor string, below), in order to ensure that `value_for_protocol` doesn't return bad results for a malformed object. Is this right?

Can you add a brief docstring saying as much, please?